addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -28,6 +29,18 @@ export class Attachment extends FileModelMixin(Record) {
     /** @type {string} */
     ownership_token;
     create_date = fields.Datetime();
+    thumbnail = fields.Attr(undefined, {
+        compute() {
+            if (
+                !this.thumbnail &&
+                this.isPdf &&
+                this.store.self_partner?.main_user_id?.share === false
+            ) {
+                this.setPdfThumbnail();
+            }
+            return this.thumbnail;
+        },
+    });
```

Comment:
This is currently a field coming from the python, it should not have a compute method on it.
Maybe onUpdate?